src/wombats/lib/wombat_lib.clj

```diff
@@ -0,0 +1,232 @@
+
+(defn get-global-state
+  "Tries to fetch global arena from the saved state or constructs a new one"
+  [state & path]
+  (let [saved (get-in state path)
+        size  (first (:global-dimensions state))]
+    (if (nil? saved)
+      (build-initial-global-state size)
+      saved )))
+
+(defn get-direction
+  "Get the current direction of your wombat from the 2d arena array"
+  [arena]
+  (get-in (nth (nth arena 3) 3) [:contents :orientation]))
+
+(defn facing?
+  "Returns true if a move forward will bring you closer to desired location
+  If no self coordinates are provided, use distance from {:x 3 :y 3}"
+  ([dir {x_tar :x y_tar :y} arena-half {x_self :x y_self :y}]
+    (case dir
+          "n" (and (not (= y_tar y_self)) (>= arena-half (mod (- y_self y_tar) (* arena-half 2)))) 
+          "e" (and (not (= x_tar x_self)) (>= arena-half (mod (- x_tar x_self) (* arena-half 2))))
+          "s" (and (not (= y_tar y_self)) (>= arena-half (mod (- y_tar y_self) (* arena-half 2))))
+          "w" (and (not (= x_tar x_self)) (>= arena-half (mod (- x_self x_tar)  (* arena-half 2))))
+          false))
+  ([dir node arena-half] (facing? dir node arena-half {:x 3 :y 3})))
+
+(defn distance-to-tile
+  "Get the number of moves it would take to move from current location.
+  If no self coordinates are provided, use distance from {:x 3 :y 3}"
+  ([dir node arena-half self-node]
+    (+ (Math/abs (- (:y node) (:y self-node)))
+       (Math/abs (- (:x node) (:x self-node)))
+       (if (facing? dir node arena-half self-node) 0 1)))
+  ([dir node arena-half]
+    (distance-to-tile dir node arena-half {:x 3 :y 3})))
+
+(defn turn-to-dir
+  "Returns one of [:right :left :about-face]"
+  [curr-dir next-dir]
+  (def ^:private orientations ["n" "e" "s" "w"])
+  (let [curr-idx (.indexOf orientations curr-dir)
+        next-idx (.indexOf orientations next-dir)]
+    (case (mod (- curr-idx next-idx) 4)
+      0  nil
+      1  :left
+      2  :about-face
+      3  :right)))
+
+(defn can-shoot-enemy?
+  "Returns true if there is a Zakano or Wombat within shooting range"
+  ([dir arena arena-size shot-range self]
+    (def shootable (case dir
+      "n" #(and (= (:x self) (:x %)) (>= shot-range (mod (- (:y self) (:y %)) arena-size)))
+      "e" #(and (= (:y self) (:y %)) (>= shot-range (mod (- (:x %) (:x self)) arena-size)))
+      "s" #(and (= (:x self) (:x %)) (>= shot-range (mod (- (:y %) (:y self)) arena-size)))
+      "w" #(and (= (:y self) (:y %)) (>= shot-range (mod (- (:x self) (:x %)) arena-size)))
+      #(false)))
+    (let [shootable (filter shootable (filter-arena arena "zakano" "wombat"))]
+      (not (empty? (filter #(not (and (= (:x %) (:x self)) (= (:y self) (:y %)))) shootable)))))
+  ([dir arena arena-size shot-range] (can-shoot-enemy? dir arena arena-size shot-range {:x 3 :y 3})))
+
+(defn can-shoot-barrier?
+  "Returns true if there is a barrier within shooting range"
+  ([dir arena arena-size shot-range self]
```

Comment:
Isn't shooting range constant? If so it might be good to have this as a constant defined up top and use it down here.